include/nbl/ui/CWindowWin32.h

```diff
@@ -16,18 +16,20 @@ class CWindowWin32 final : public IWindowWin32
 	static LRESULT CALLBACK WndProc(HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam);
 
 public:
-	explicit CWindowWin32(core::smart_refctd_ptr<system::ISystem>&& sys, native_handle_t hwnd) : IWindowWin32(std::move(sys)), m_native(hwnd)
+	//TODO Do we even want to have this ctor now since the window should be created in a separate thread??
```

Comment:
yes, it's for integration with external frameworks (window handle created by them)

Reply:
will see, leave for now, but dont bother with implementation

Reply:
make sure you `assert(false)` in functions that are not supposed to work/be called/left as TODO

Reply:
maybe make ctor protected?